dsl/graphql/interaction.go

```diff
@@ -0,0 +1,108 @@
+package graphql
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/pact-foundation/pact-go/dsl"
+)
+
+// Variables represents values to be substituted at runtime
+type Variables map[string]interface{}
+
+// Request is the main implementation of the Pact interface.
+type Request struct {
```

Comment:
It will be useful to add `Extensions` to this struct so that requests can be made with extensions for using feature like [APQ](https://www.apollographql.com/docs/apollo-server/performance/apq/).  

`Extensions map[string]interface{}`